include/nds/arm9/videoGL.h

```diff
@@ -794,7 +794,7 @@ static inline void glColor(rgb color)
 ///     The z component for the vertex.
 static inline void glVertex3v16(v16 x, v16 y, v16 z)
 {
-    GFX_VERTEX16 = (y << 16) | (x & 0xFFFF);
+    GFX_VERTEX16 = ((u32)(u16)y << 16) | (x & 0xFFFF);
```

Comment:
I used u32/u16 here instead of the standard types uint32_t/uint16_t like below because the parameters also use the custom types.